src/components/Chat/ChatInterface.tsx

```diff
@@ -82,97 +115,115 @@ export default function ChatInterface() {
-            setMsgs(
-                [
-                    <MessageOther
-                        message={res.message}
-                        timestamp=""
-                        displayName="AI Tutor"
-                        avatarDisp={true}
-                    />,
-                    ...newRenderedMsgs
-                ]
-            );
-
-            setRawMsgs(newMsgs);
-
-            setShowSpinner(false);
-
-            if (!res['current_step']) {
-                alert('Error: current_step not found in response.');
-                console.error(res);
-            } else {
-                if (res['current_step'] !== currentStep) {
-                    setCorrectSoFar(0);
-                    setQuestions(undefined);
-                }
-                setCurrentStep(res['current_step']);
-            }
 
-            if (res['correct'] === true) {
-                setCorrectSoFar(correctSoFar + 1);
-            }
-
-            if (res['questions']) {
-                setQuestions(res['questions']);
-            }
-            
+        if (useStreaming) {
+            req.history = getHistory(newMsgs);
+
+            const newRenderedMsgs = [
+                <MessageSelf
+                    message={msg}
+                    timestamp=""
+                    displayName="User"
+                    avatarDisp={false}
+                />,
+                ...msgs
+            ]
+
+            setMsgs(newRenderedMsgs);
+
+            getResponse(req, lastQuestion ?? '', setRawResponseData)
+                .then((res) => {
+                    if ('tutor_response' in res && res.tutor_response !== '') {
+                        const levelUndecided = questionLevel === undefined;
+                        const msg = renderTutorResponseFinal(res.tutor_response, res.follow_up_question, res.question_completed, levelUndecided);
+                        console.log("msg: ", msg)
+
+                        const [msgElems, msgData] = updateMsgList(msg, 'AI Tutor', newMsgs, newRenderedMsgs, setMsgs, setRawMsgs);
+
+                        setRawResponseData([]);
+
+
+                        let newCorrectSoFar = correctSoFar;
+
+                        // track correctness if current question is completed
+                        // then the useEffect hook will show the next question on the level if threshold has not been met
+                        if (questionLevel && res.question_completed === "true") {
+                            newCorrectSoFar += 1;
+                            setCorrectSoFar(newCorrectSoFar);
+                        }
+
+                        // update current level if correctSoFar reaches threashold
+                        if (newCorrectSoFar === THRESHOLD) {
+                            const idxCurrent = bloomsTaxonomy.indexOf(questionLevel);
+                            const idxNew = idxCurrent + 1;
+                            if (questionLevel === 'analyze' || idxNew === bloomsTaxonomy.length) {
```

Comment:
questionLevel might not be in bloomsTaxonomy because it is the AI's judgment. So far the non-taxonomy values I have seen include empty strings and `none`. Any thoughts on how to address this?